tests/test_cli_products_generate.py

```diff
@@ -1,45 +1,178 @@
-# Copyright 2024 Broda Group Software Inc.
-#
-# Use of this source code is governed by an MIT-style
-# license that can be found in the LICENSE file or at
-# https://opensource.org/licenses/MIT.
-#
-# Created:  2024-04-15 by eric.broda@brodagroupsoftware.com
-import pytest
-import os
 import json
 
 from cli import main
 import state
 
-# Global state variables
-
 # Common for all CLI commands
 HOST="localhost"
 PORT="20000"
-SAMPLES_DIR = os.environ.get("SAMPLES_DIR")
-DATAPRODUCT_DIR=os.path.join(SAMPLES_DIR, "dataproducts", "rmi")
 
+# NAMESPACE="brodagroupsoftware.com" ;
+# NAME="rmi.dataproduct" ;
+# TAGS="utilities,emissions" ;
+# DESCRIPTION="Some short description" ;
+# DATA_URL="Some data url" ;
+# OUTPUT_DIRECTORY="./output" ;
+# PRODUCT_YAML="product.yaml" ;
+# python ./src/cli.py $VERBOSE --host $HOST --port $PORT products \
+#     --generate \
+#     --namespace "$NAMESPACE" \
+#     --name "$NAME" \
+#     --tags "$TAGS" \
+#     --data "$DATA_URL" \
+#     --description "$DESCRIPTION" \
+#     --directory "$OUTPUT_DIRECTORY" \
+#     --filename "$PRODUCT_YAML"
+def test_generate_product_manual():
+    TAGS = "tag1,tag2,tag3"
+    NAMESPACE = "test-space"
+    NAME = "test-name"
+    DIRECTORY = "./output"
+    FILENAME = "test_product.yaml"
+    DESCRIPTION = "A short description"
+    result = main([
+        "--host", HOST,
+        "--port", PORT,
+        "products",
+        "--generate",
+        "--namespace", NAMESPACE,
+        "--name", NAME,
+        "--tags", TAGS,
+        "--description", DESCRIPTION,
+        "--directory", DIRECTORY,
+        "--filename", FILENAME
+    ])
+    assert(result is not None)
+    result = json.loads(result)
+    assert "product" in result
+
+# NAMESPACE="brodagroupsoftware.com" ;
+# NAME="rmi.dataproduct" ;
+# TAGS="utilities,emissions" ;
+# URL="https://data.catalyst.coop/pudl" ;
+# OUTPUT_DIRECTORY="./output" ;
+# PRODUCT_YAML="product.yaml" ;
+# VENDOR="OpenAI" ;
+# MODEL="gpt-3.5-turbo" ;
+# python ./src/cli.py $VERBOSE --host $HOST --port $PORT products \
+#     --generate \
+#     --namespace "$NAMESPACE" \
+#     --name "$NAME" \
+#     --tags "$TAGS" \
+#     --url "$URL" \
+#     --vendor "$VENDOR" \
+#     --model "$MODEL" \
+#     --directory "$OUTPUT_DIRECTORY" \
+#     --filename "$PRODUCT_YAML"
+def test_generate_product_llm():
```

Comment:
ai should probably either be mocked, or not be in unit tests, as it makes it very difficult to automated, or for other people to run.

Reply:
moved to integration tests